sklearn/neighbors/_binary_tree.pxi.tp

```diff
@@ -2111,16 +2104,18 @@ cdef class BinaryTree{{name_suffix}}:
                 node_log_min_bounds[i1] = (log(N1) +
                                            compute_log_kernel(dist_UB_1,
                                                               h, kernel))
-                node_log_bound_spreads[i1] = (log(N1) +
-                                              compute_log_kernel(dist_LB_1,
-                                                                 h, kernel))
+                node_log_bound_spreads[i1] = logsubexp(
```

Comment:
Do you think that we could have this fix in another PR. I'm really not comfortable with this part of the code and I think that this PR is fixing two bugs in one.

We could fix first the bandwidth estimation. Regarding this change, do you have any additional resource explaining the algorithm to understand what is wrong?

Reply:
Ah sure it is some early trials that I forgot to revert. I will update the top comment with the latest results as well.

Reply:
These removed :)